### ISSUE TITLE
feature: Improve handling of whitespace in tables with links in their cells

### ISSUE THREAD

**Issue by BaumerEDV** [MeanderingProgrammer/render-markdown.nvim#175]
### Is your feature request related to a problem? Please describe.

The issue arises when working with a table that contains links of varying size in the markdown code **and** the user elects to vertically align table columns in the code, such as here:
![image](https://github.com/user-attachments/assets/9f15f37b-aca0-47aa-ba23-28aebeea249f)
which then renders as:
![image](https://github.com/user-attachments/assets/b4445eb1-a0a9-4e2f-9e64-520a1bf574ea)
This wastes a lot of screen space on white space and makes the table hard to read.

### Describe the solution you'd like

It would be preferable if the table rendered as such, with the padding stripped:
![image](https://github.com/user-attachments/assets/9df37a9b-cd35-451b-ba04-91ed0face9c7)

### Describe alternatives you've considered

The user could elect to not vertically align the pipes of the columns. But this makes the markdown document more difficult to use for other people viewing it in plain text (such as coworkers not using this plugin).
![image](https://github.com/user-attachments/assets/fb0942fa-c2e1-458b-8b0c-c69db48f6915)

### Additional information

The relevant section of the table for your convenience:
```
| Points To                                                                                                                                                                    | Level   |
| ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------- |
| [[TTRPG/WWN/Caeraema/Locations/Pike Cave/_Pike Cave\|Pike Cave]]                                                                                                             | Low     |
| [[TTRPG/WWN/Caeraema/Locations/Floria's Tower/_Floria's Tower\|Floria's Tower]]                                                                                              | Low     |
| [[TTRPG/WWN/Caeraema/Locations/The Sunken Fane/_The Sunken Fane\|The Sunken Fane]]                                                                                           | Low     |
| [[TTRPG/WWN/Caeraema/Locations/The Sunken Fane/_The Sunken Fane\|The Sunken Fane]]                                                                                           | Low     |
| [[TTRPG/WWN/Caeraema/Locations/The Sunken Fane/_The Sunken Fane\|The Sunken Fane]]                                                                                           | Low     |
| [[TTRPG/WWN/Caeraema/Locations/Ruins of Stormwatch/_Ruins of Stormwatch\|Ruins of Stormwatch]]                                                                               | Low     |
| [[_The Haunted Mill\|The Haunted Mill]]                                                                                                                                      | Low     |
| [[_The Whispering Hollow]]                                                                                                                                                   | Low     |
| [[_The Whispering Hollow]]                                                                                                                                                   | Low     |
| [[_The Logging Venture]]                                                                                                                                                     | Low     |
| [[_Marauder Cave\|Marauder Cave]]                                                                                                                                            | Low     |
| [[_Owl's Roost\|Owl's Roost]]                                                                                                                                                | Low     |
| [[_Blighted Orchard]]                                                                                                                                                        | Low     |
| [[_Gloomwood Asylum\|Gloomwood Asylum]]                                                                                                                                      | Low     |
| [[_Witchfire Plains\|Witchfire Plains]]                                                                                                                                      | Low/Mid |
| [[_The Bloodpit]]                                                                                                                                                            | Low/Mid |
| [[_Titan Shrine\|Titan Shrine]]                                                                                                                                              | Low     |
| [[_Alatryn's Tower\|Alatryn's Tower]]                                                                                                                                        | Low     |
| [[_Passing Forest Chapel\|Passing Forest Chapel]]                                                                                                                            | Low     |
| [[TTRPG/WWN/Caeraema/Locations/Serpent's Isle/_Serpent's Isle\|Serpent's Isle]]                                                                                              | Mid     |
| [[TTRPG/WWN/Caeraema/Locations/Serpent's Isle/_Serpent's Isle\|Serpent's Isle]]                                                                                              | Mid     |
| [[_Goblin-Kin Caves]]                                                                                                                                                        | Mid     |
| [[_Jungle Fever Hollow\|Jungle Fever Hollow]]                                                                                                                                | Mid     |
| [[_Necrotic Glade]]                                                                                                                                                          | Mid     |
| [[_Quicksilver Pool\|Quicksilver Pool]]                                                                                                                                      | Mid     |
| [[_Toxic Fungus Hollow\|Toxic Fungus Hollow]]                                                                                                                                | Mid     |
| [[_Mangrove Cove]]                                                                                                                                                           | Mid     |
| [[_Bog of Eternal Rain]]                                                                                                                                                     | Mid     |
| [[_Bloodwood Grove]]                                                                                                                                                         | Mid     |
| [[_Shipwreck Cove]]                                                                                                                                                          | Mid     |
| [[_Jungle Sinkhole]]                                                                                                                                                         | Mid     |
| [[_Wyrmfang Bluff]]                                                                                                                                                          | Mid     |
| [[_Greybeard's Rest]]                                                                                                                                                        | Any     |
| [[_Soulfire Crag]]                                                                                                                                                           | Mid     |
| [[_Serpent's Gorge]]                                                                                                                                                         | Mid     |
| [[_Starfall Observatory]]                                                                                                                                                    | Any     |
| [[_Bloodfang Caverns]]                                                                                                                                                       | Mid     |
| [[_Dwarven Bridge Ruins]]                                                                                                                                                    | Mid     |
| [[_Dwarven Lighthouse]]                                                                                                                                                      | Mid     |
| [[_Golem Graveyard]]                                                                                                                                                         | Mid     |
| [[_Thunderpeak Plateau]]                                                                                                                                                     | Mid     |
| [[_Moonshadow Plateau]]                                                                                                                                                      | Mid     |
| [[_Bonegrinder Peak]]                                                                                                                                                        | Mid     |
| [[TTRPG/WWN/Caeraema/Locations/Thaladrimarixtur, The Eternal Blizzard's Lair/_Thaladrimarixtur, The Eternal Blizzard's Lair\|Thaladrimarixtur, The Eternal Blizzard's Lair]] | High    |
```

I recognize that this issue can be somewhat mitigated by having shorter links and then deleting the whitespace and ---s causing a lot of the padding, but this issue would also exist for http hyperlinks which cannot necessarily be shortened in this way.

Thank you for this terrific plugin and I hope you have a wonderful day <3

**Comment by MeanderingProgrammer** [comment]
I'll see what I can do, will likely be a new cell style if it's not too much of a pain to implement.

**Comment by MeanderingProgrammer** [comment]
Added a new cell value `trimmed` here: https://github.com/MeanderingProgrammer/render-markdown.nvim/commit/c68697085441d03a20eee15d4d78e2e5a771569a

That should handle this like you describe. Update the plugin and set the configuration like so:j

```lua
require('render-markdown').setup({
    pipe_table = { cell = 'trimmed' },
})
```

**Comment by BaumerEDV** [comment]
Works like a charm

I verified it with the table above, as well as with tables that place the links in the not-first column and found no problems.
Furthermore, I tried to break it by removing part of the whitespace / the header ---s in some lines / unaligning the column pipes; the table rendered correctly each time.
The documentation for the feature looks good / is understandable to me as well.

As far as I am concerned, this feature request is completed.

Thank you very much for the change!
Is there a way to buy you a coffee?

In any case cheers and enjoy your weekend!
